package.json

```diff
@@ -172,6 +173,7 @@
   },
   "resolutions": {
     "@types/node": "~14.14.45",
+    "@types/react": "^18.2.21",
```

Comment:
This is the only way I was able to avoid TS issues. Other alternatives can be explored (reduce hoisting etc) but that looks fine to me?